docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,865 @@
+
+Process and Product Design Principles: Synthesis, Analysis, and Evaluation. Seider, Seader, Lewin, Windagdo, 3rd Ed. John Wiley and Sons. Chapter 22. Cost Accounting and Capital Cost Estimation
+
+Users should refer to the reference above for details of the costing correlations, however, a summary of this methods is provided below.
+    
+Table 2. Cost basis for each unit model.
+
+=========================== =========================  ===========
+Unit Model                  Basis                      Units      
+=========================== =========================  ===========
+heat exchanger              :math:`area`               ft^2       
+pump                        :math:`fluid_{work}`       ft^3/s     
+compressor                  :math:`mechanical_{work}`  hp         
+turbine                     :math:`mechanical_{work}`  hp         
+vessels                     :math:`D , L`              ft         
+fired heaters               :math:`heat\_duty`         BTU/hr         
+=========================== =========================  ===========
+
+Unit Mapping
+^^^^^^^^^^^^
+
+As a convenience for users, common unit models are mapped with appropriate costing methods. Therefore, for quick costing with defaults, users may import the `unit_mapping` dictionary and pass the unit model type as below:
+
+.. code:: python
+
+    unit_mapping = {  
+            CSTR: cost_vertical_vessel,  
+            Compressor: cost_compressor,  
+            Flash: cost_vertical_vessel,  
+            Heater: cost_fired_heater,  
+            HeatExchanger: cost_heat_exchanger,  
+            HeatExchangerNTU: cost_heat_exchanger,  
+            PFR: cost_horizontal_vessel,  
+            PressureChanger: cost_pressure_changer,  
+            Pump: cost_pump,  
+            StoichiometricReactor: cost_horizontal_vessel,  
+            Turbine: cost_turbine,  
+        }
+
+    costing_method = unit_mapping[type(blk)]
+
+The mapping is inheritance aware, e.g. pumps will try costing with `cost_pump()` and fall back to `cost_pressure_changer()`.
+
+For example, using default unit mapping the heat exchanger example would add costing as:
+
+.. code:: python
+
+    m.fs.H101.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.costing,
+            "costing_method": SSLWCostingData.unit_mapping[HeatExchanger],  # passing the imported class object
+            "costing_method_arguments": {
+                    "material_type": HeaterMaterial.CarbonSteel,
+                    "heat_source": HeaterSource.Fuel
+            }
+        }
+    )
+
+The unit mapping is particularly useful when defining costing for a group of unit models with similar costing arguments and variables, such as vessels, by only passing the model class and not a specific costing method. In the code below, a flowsheet containing two `Flash` vessels, a `CSTR` unit and a `StoichiometricReactor` may be costed compactly via a loop:
+
+.. code:: python
```

Comment:
I think you should drop this example. Firstly, it is misusing the mapping, and secondly I am not sure what you are trying to show (it needs a lot more explanation to be meaningful).